scripts/main/view.js

```diff
@@ -482,51 +518,76 @@ view.album = {
 						: ratio;
 				});
 
+				/**
+				 * An album listing has potentially hundreds of photos, hence
+				 * only query for them once.
+				 * @type {jQuery}
+				 */
+				const jqPhotoElements = $(".justified-layout > div.photo");
+				const photoDefaultHeight = parseFloat(jqPhotoElements.css("--lychee-default-height"));
+
 				const layoutGeometry = require("justified-layout")(ratio, {
 					containerWidth: containerWidth,
 					containerPadding: 0,
-					// boxSpacing: {
-					//     horizontal: 42,
-					//     vertical: 150
-					// },
-					targetRowHeight: parseFloat($(".photo").css("--lychee-default-height")),
+					targetRowHeight: photoDefaultHeight,
 				});
-				// if (lychee.rights.is_admin) console.log(layoutGeometry);
-				$(".justified-layout").css("height", layoutGeometry.containerHeight + "px");
-				$(".justified-layout > div").each(function (i) {
+				// Temporarily hide the container such that not each
+				// modification of every photo triggers a UI update.
+				jqJustifiedLayout.addClass("relayouting");
+				// We must give the UI worker a chance to apply the class
+				// above, hence the remaining part of this method must be
+				// done asynchronously.
```

Comment:
Am I missing something? I don't see the reminder of this method being done asynchronously...

Reply:
Fixed in https://github.com/LycheeOrg/Lychee-front/pull/334/commits/c7a780b47f15e5aa126db8d633748c8873ea659b.